backend/layers/persistence/persistence.py

```diff
@@ -679,11 +686,16 @@ def finalize_collection_version(
             # get all dataset versions for the datasets that are being tombstoned
             dataset_version_ids_to_delete_from_s3 = []
             if dataset_ids_to_tombstone:
-                datasets = session.query(DatasetTable).filter(DatasetTable.id.in_(dataset_ids_to_tombstone)).all()
-                for dataset in datasets:
-                    dataset.tombstone = True
-                    dataset_all_versions = session.query(DatasetVersionTable).filter_by(dataset_id=dataset.id).all()
-                    dataset_version_ids_to_delete_from_s3.extend([dv.id for dv in dataset_all_versions])
+                tombstone_dataset_statement = (
+                    update(DatasetTable).where(DatasetTable.id.in_(dataset_ids_to_tombstone)).values(tombstone=True)
+                )
+                session.execute(tombstone_dataset_statement)
+                dataset_all_version_ids = (
```

Comment:
the tests are passing so maybe i'm misunderstanding, but this query looks like we're getting List[DatasetVersion] rather than List[DatasetVersionId], no? And dataset_version_ids_to_delete_from_s3 expects Ids

Reply:
You're totally right, but that is how the original code was written. I change it to query for DatasetVersion.id and it works. Not sure how it managed to work before, because it was returning a row not a string.